crates/lair_keystore_api/src/ipc_keystore.rs

```diff
@@ -0,0 +1,302 @@
+
+    /// Get the config used by the LairServer held by this IpcKeystoreServer.
+    pub fn get_config(&self) -> LairServerConfig {
+        self.config.clone()
+    }
+}
+
+/// Client connection options for customizing how to connect to an ipc server.
+pub struct IpcKeystoreClientOptions {
+    /// the ipc url ('unix://' or 'named-pipe:\\.\pipe\[yada]') to connect to
+    pub connection_url: url::Url,
+    /// the passphrase to use to connect
+    pub passphrase: sodoken::BufRead,
+    /// Require the client and server to have exactly matching
+    /// client / server versions.
+    pub exact_client_server_version_match: bool,
+    /// If it is not possible to unlock the server "interactively" on
+    /// the server side... we can unlock it from the client side without
+    /// validating the server's authenticity...
+    /// If the server is NOT authentic, we will get an error, but we
+    /// also will have supplied our passphrase to an attacker.
+    pub danger_unlock_without_server_validate: bool,
+}
+
+/// Connect to an IpcKeystoreServer instance via
+/// unix domain socket on linux/macOs or named pipe on windows.
+/// This constructor will first validate server authenticity,
+/// then unlock the connection with the supplied passphrase.
+pub fn ipc_keystore_connect<P>(
+    connection_url: url::Url,
+    passphrase: P,
+) -> impl Future<Output = LairResult<LairClient>> + 'static + Send
+where
+    P: Into<sodoken::BufRead> + 'static + Send,
+{
+    let passphrase = passphrase.into();
+    ipc_keystore_connect_options(IpcKeystoreClientOptions {
+        connection_url,
+        passphrase,
+        exact_client_server_version_match: false,
+        danger_unlock_without_server_validate: false,
+    })
+}
+
+/// Connect to an IpcKeystoreServer instance via
+/// unix domain socket on linux/macOs or named pipe on windows.
+pub fn ipc_keystore_connect_options(
+    opts: IpcKeystoreClientOptions,
+) -> impl Future<Output = LairResult<LairClient>> + 'static + Send {
+    async move {
+        let server_pub_key =
+            get_server_pub_key_from_connection_url(&opts.connection_url)?;
+
+        // establish the raw ipc connection
+        let (send, recv) =
+            raw_ipc::ipc_connect(opts.connection_url.clone()).await?;
+
+        // wrap this connection up as a LairClient
+        let cli_hnd =
+            crate::lair_client::async_io::new_async_io_lair_client(send, recv)
+                .await?;
+
+        if opts.danger_unlock_without_server_validate {
+            // even if they tell us to do this backwards,
+            // let's at least try to do it correctly to start
+            match cli_hnd.hello(server_pub_key.clone()).await {
+                Ok(ver) => {
+                    priv_check_hello_ver(&opts, &ver)?;
+
+                    // hey, it worked, unlock and proceed
+                    cli_hnd.unlock(opts.passphrase.clone()).await?;
+                }
+                Err(e) if e.str_kind() == "KestoreLocked" => {
```

Comment:
typo?

Reply:
yep! : ) - I've got a backlog task to normalize these kinds to an enum instead of strings...